engine/src/main/java/org/hibernate/validator/path/PropertyNode.java

```diff
@@ -0,0 +1,20 @@
+/*
+ * Hibernate Validator, declare and validate application constraints
+ *
+ * License: Apache License, Version 2.0
+ * See the license.txt file in the root directory or <http://www.apache.org/licenses/LICENSE-2.0>.
+ */
+package org.hibernate.validator.path;
+
+/**
+ * Node representing a property, providing Hibernate Validator specific functionality.
+ *
+ * @author Gunnar Morling
+ */
+public interface PropertyNode extends javax.validation.Path.PropertyNode {
```

Comment:
nice to create the public class for this (compared to just exposing something internal). I think we get really good at this.